waku/v2/protocol/rln/group_manager/dynamic/dynamic.go

```diff
@@ -66,12 +66,12 @@ func handler(gm *DynamicGroupManager, events []*contracts.RLNMemberRegistered) e
 	toInsertTable := om.New()
 	for _, event := range events {
 		if event.Raw.Removed {
-			var indexes []uint64
+			var indexes []uint
 			i_idx, ok := toRemoveTable.Get(event.Raw.BlockNumber)
 			if ok {
-				indexes = i_idx.([]uint64)
+				indexes = i_idx.([]uint)
 			}
-			indexes = append(indexes, event.Index.Uint64())
+			indexes = append(indexes, uint(event.Index.Uint64()))
```

Comment:
nvm. Looking at the implementation in zerokit, the functions that accept indexes receive an `uintptr_t` value which will depend on the platform architecture...

@rymnc - do you think we should change zerokit's FFI api to use u64 instead of usize for the indexes? what's the maximum number of leaves that a merkle tree can have?

Reply:
by default, the tree_height is 20, so `1_048_576`
I think we fixed the issue of  serialization for usize's 🤔

Reply:
indeed, serialization should be fine, but i wonder if in this case (ffi), it would be a problem, as you'd be limited to a maximum index of 2^32-1 in a 32b machine, due to using usize :thinking:

Reply:
I don't believe we'd cross 2^32 ~ since 2^20 is the max tree height